### DISCUSSION TITLE
How to prevent users from copying and pasting source code externally？

### DISCUSSION THREAD

**Discussion by ColaemmZuo** [coder/code-server#6528] [Q&A] [answered]
Since our company's business has high security requirements for source code, we hope that users will not be able to download or copy and paste the source code to the outside.

I already know downloads can be disabled via `--disable-file-downloads`.

Is there a way to prohibit copying and pasting source code? Thanks in advance!

**Comment by code-asher** [comment, marked as answer]
There is no mechanism to prevent copy/paste currently.  If someone wants to tackle it, PRs are welcome!

**Comment by BYTESHAUS** [comment]
If the code is open even not publically but to a wide range of users outside the company, you can't prevent it.
Copyright and policies don't work for some locations (known for copy-paste in technics). That's why you never see the bread-winning algorithms (like graphs) anywhere.
